File: async_function_and_await.js
import { setTimeout } from 'timers/promises';
// This setTimeout function is a promise-based version of the traditional setTimeout, which allows us to use it with async/await syntax.

// With a blocking code block.
async function async_function(){
    await setTimeout(2000)
    console.log("Hello from blocking code block")
}

// With a non-blocking code block.
async function async_function2(){
    console.log("Hello from non-blocking code block")
}

// To run asynchronous code in python, you need a runner
await async_function2()
await async_function()

File: async_function_and_await.py
import asyncio

# With a blocking code block.
async def async_function():
    await asyncio.sleep(2)
    print("Hello from blocking code block")

# With a non-blocking code block.
async def async_function2():
    print("Hello from non-blocking code block")

# To run asynchronous code in python, you need a runner
asyncio.run(async_function2())
asyncio.run(async_function())


# Its simple, if there's any asynchronous code
# Keep it in runner.